vortex-layout/src/layouts/file_stats.rs

```diff
@@ -42,12 +42,25 @@ pub struct FileStatsAccumulator {
 impl FileStatsAccumulator {
     fn new(dtype: &DType, stats: Arc<[Stat]>, max_variable_length_statistics_size: usize) -> Self {
         let accumulators = Arc::new(Mutex::new(match dtype.as_struct() {
-            Some(dtype) => dtype
-                .fields()
-                .map(|field_dtype| {
-                    StatsAccumulator::new(&field_dtype, &stats, max_variable_length_statistics_size)
-                })
-                .collect(),
+            Some(struct_dtype) => {
+                if dtype.nullability() == Nullability::Nullable {
+                    vortex_panic!(
+                        "FileStatsAccumulator does not support nullable top-level structs, got: {}. Use Validity::NonNullable",
+                        dtype
+                    );
+                }
```

Comment:
Maybe add a comment say this is a temporary fix until we support files with top level nullability